msger_mosaic_bridge/src/conn_manager.cpp

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (C) 2024 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+
+#include "conn_manager.hpp"
+#include "udp_listener.hpp"
+
+bool ConnectionManager::connect(const std::string &connectionType,
```

Comment:
Just to confirm my understanding, there's ONE shared connection manager instance, which manages multiple different sockets which are referenced by their connectionType value?

I'm not a huge fan of using strings for that, as a rule of thumb. It'll function fine, and at this scale it's probably not a huge deal. But it might be unclear to later users (or people only looking at one section of the code) what different values of connection type are used where. For example, it's hard to know if I'm extending this whether the connection is called "vehicle_state" or "vehicle_status", among other kinds of errors that could occur (typos,etc.).

Making an enum of connection types would be one way to solve this, making it more clear what kinds of connections are intended to be supported, and precisely how to identify them. Alternatively, a more complicated solution would be to have this connect function return some sort of reference to an underlying connection object such that no explicit ID value is needed. e.g.:

```auto vehicle_status_connection = conn_mgr_.connect(...)``` and the actual variable name `vehicle_status_connection` serves as a unique identifier for it, if that makes sense.

Bit of a nitpick, but if you can take a minute to think on it and then decide whether or not you want to proceed with the current implementation, I'm ultimately fine with your judgment either way.

Reply:
Added enum